src/examples/using-with-cmake/CMakeLists.txt

```diff
@@ -70,6 +70,14 @@ if(BLT_CMAKE_IMPLICIT_LINK_DIRECTORIES_EXCLUDE)
                      ${BLT_CMAKE_IMPLICIT_LINK_DIRECTORIES_EXCLUDE})
 endif()
 
+# HACK: Define the `mpi` target as an alias to `axom::mpi` if/when necessary
+# Details: An axom dependency has a link dependency to `mpi`, but it might
+# not be a target. Need to mark axom::mpi as global to use it for an alias.
+if(NOT TARGET mpi AND TARGET axom::mpi)
+    set_target_properties(axom::mpi PROPERTIES IMPORTED_GLOBAL TRUE)
+    add_library(mpi ALIAS axom::mpi)
+endif()
```

Comment:
@white238 -- this fixes the problem, but doesn't seem like the "right" approach. Do you have a better suggestion?

Reply:
Notes:
* The problem was that one of axom's dependencies (`conduit_relay`, I think) was adding `-lmpi` to the link line, but `mpi` was not a CMake target.
* A non-global CMake target can be aliased in CMake@3.18+, but this example only depends on CMake@3.14, so I modified the "IMPORTED_GLOBAL" property.

Reply:
This seems odd to me. Are you mixing CMake versions? The find_dependency should have created all these targets and called Conduit's config which creates their necessary targets. This only happens to me if I do the build with a different CMake then testing the CMake using example.

Reply:
Could be -- I was using axom's `config-build` for the initial build and whatever was in my path for the `using-with-cmake` example.

If that's the issue, we should probably keep the fix, but update the comment?

Reply:
**Update:** The linker error was still there without this bugfix if I use the same version of cmake to configure axom and the `using-with-cmake` example.

```shell
> grep -i cmake  <axom>/host-configs/rzgenie-toss_3_x86_64_ib-clang@10.0.0.cmake 
# CMake executable path: /usr/tce/packages/cmake/cmake-3.21.1/bin/cmake

...

> cd <axom>/install-axom/examples/axom/using-with-cmake/build
> /usr/tce/packages/cmake/cmake-3.21.1/bin/cmake -C ../host-config.cmake ..

loading initial cache file ../host-config.cmake
-- The CXX compiler identification is Clang 10.0.0
...
-- CMake Version: 3.21.1
...

> make
[ 50%] Building CXX object CMakeFiles/example.dir/example.cpp.o
[100%] Linking CXX executable example
/usr/bin/ld: cannot find -lmpi
clang-10: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [example] Error 1
make[1]: *** [CMakeFiles/example.dir/all] Error 2
make: *** [all] Error 2
```